main.go

```diff
@@ -147,7 +174,78 @@ func (fp *Firehose) handleMessage(ctx context.Context, message *hedwig.Message)
 	return err
 }
 
-func (fp *Firehose) RunLeader() {
+func (fp *Firehose) RunLeader(ctx context.Context) error {
+	currentTime := fp.clock.Now()
+	timerCh := time.After(time.Duration(fp.processSettings.ScrapeInterval) * time.Second)
+	groupedMsgs := map[string]byTimestamp{}
+	// go through all msgs and write to msgtype folder
+	for {
+		select {
+		case <-timerCh:
+			// read from staging
+			fileNames, err := fp.storageBackendCreator.ListFiles(ctx, fp.processSettings.StagingBucket)
+			if err != nil {
+				return err
+			}
+			// sort by topic
```

Comment:
would it make sense to write files in a heirarchy in the staging bucket like `<topic-name>/<version>/<time-window-bucket>.firehose` so we can also just read per topic here?